mantle/cmd/ore/gcloud/deprecate-image.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcloud
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/coreos/mantle/platform/api/gcloud"
+	"github.com/spf13/cobra"
+)
+
+var (
+	cmdDeprecateImage = &cobra.Command{
+		Use:   "deprecate-image --image-name=ImageName [--state=DeprecationState]",
+		Short: "Deprecate GCP image",
+		Long:  "Change deprecation status of existing GCP image",
+		Run:   runDeprecateImage,
+	}
+
+	deprecateImageName        string
+	deprecateImageState       string
+	deprecateImageReplacement string
+)
+
+func init() {
+	cmdDeprecateImage.Flags().StringVar(&deprecateImageName,
+		"image-name", "", "GCP image name")
+	cmdDeprecateImage.MarkFlagRequired("image-name")
+	cmdDeprecateImage.Flags().StringVar(&deprecateImageState, "state",
+		string(gcloud.DeprecationStateDeprecated),
+		fmt.Sprintf("Deprecation state must be one of: %s,%s,%s,%s",
+			gcloud.DeprecationStateActive,
+			gcloud.DeprecationStateDeprecated,
+			gcloud.DeprecationStateObsolete,
+			gcloud.DeprecationStateDeleted))
+	cmdDeprecateImage.Flags().StringVar(&deprecateImageReplacement,
+		"replacement", "", "optional: link to replacement for the deprecated image")
+	GCloud.AddCommand(cmdDeprecateImage)
+}
+
+func runDeprecateImage(cmd *cobra.Command, args []string) {
+
+	// Check that the deprecation state is a valid one
+	switch gcloud.DeprecationState(deprecateImageState) {
+	case gcloud.DeprecationStateActive,
+		gcloud.DeprecationStateDeprecated,
+		gcloud.DeprecationStateObsolete,
+		gcloud.DeprecationStateDeleted:
+		// Do nothing, state is valid
+	default:
+		fmt.Fprintf(os.Stderr,
+			"Specified deprecation state is invalid: %s", deprecateImageState)
+		os.Exit(1)
+	}
+
+	fmt.Printf("Attemping to change GCP image deprecation state of %s to %s\n",
```

Comment:
This probably fits better as a `plog.Debugf` statement to gate it behind the user requesting more verbose output.

Reply:
ok I switched this to `plog.Debugf` and also the other statements in the file to `plog.Fatal*`. The one thing, I can't figure out how to get the debug statement to print out. I've tried `--debug`, `--log-level=DEBUG`, and `--verbose`.

Reply:
OOB IRC discussion happened but looks like you've found a bug: #1321